src/main/java/usi/si/seart/job/CrawlProjectsJob.java

```diff
@@ -156,7 +156,7 @@ private void retrieveRepos(Range<Date> dateRange, String language, Boolean crawl
                     JsonArray results = result.get("items").getAsJsonArray();
                     saveRetrievedRepos(results, language, 1, totalResults);
                     retrieveRemainingRepos(dateRange, language, crawlUpdatedRepos, totalPages);
-                    crawlJobService.updateCrawlDateForLanguage(language, dateRange.upperEndpoint());
+                    crawlJobService.updateCrawlDateForLanguage(language, dateRange.lowerEndpoint());
```

Comment:
This was a bugfix I independently committed straight to `main`, make sure you don't use the faulty code.